openapi.yml

```diff
@@ -0,0 +1,185 @@
+      env:
+        default: prod
+  - url: 'https://api.{env}.sinopia.io'
+    description: Staging service
+    variables:
+      env:
+        default: stage
+paths:
+  /:
+    get:
+      summary: A healthcheck endpoint
+      responses:
+        '200':
+          description: The status of the service
+  /resource/{resourceId}:
+    get:
+      summary: Return a JSON representation of the given resource
+      responses:
+        '200':
+          description: OK - returns resource
+          content:
+            application/json:
+              schema:
+                type: string
+        '404':
+          description: Not found
+      parameters:
+        - name: resourceId
+          in: path
+          description: ID of the resource
+          required: true
+          schema:
+            type: string
+    post:
+      summary: Endpoint to insert or update resources - returns the resource as JSON
+      responses:
+        '200':
+          description: OK - returns resource
+          content:
+            application/json:
+              schema:
+                type: string
+        '404':
+          description: Not found
+      parameters:
+        - name: resourceId
+          in: path
+          description: ID of the resource
+          required: true
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+    put:
+      summary: Endpoint to insert or update resources - returns the resource as JSON
+      responses:
+        '200':
+          description: OK - returns resource
+          content:
+            application/json:
+              schema:
+                type: string
+        '404':
+          description: Not found
+      parameters:
+        - name: resourceId
+          in: path
+          description: ID of the resource
+          required: true
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+  /resource/{resourceId}/versions:
+    get:
+      summary: Return the resources
+      responses:
+        content:
+          application/json:
+            schema:
+              type: string
+  parameters:
+        - name: resourceId
+          in: path
+          description: ID of the resource
+          required: true
+          schema:
+            type: string
+  /resource/{resourceId}/version/{timestamp}:
+    get:
+      summary: Return the resources
+      responses:
+      parameters:
+        - name: resourceId
+          in: path
+          description: ID of the resource
+          required: true
+          schema:
+            type: string
+        - name: timestamp
+          in: path
+          description: Timestamp (version) of the resource to retrieve
+          required: true
+          schema:
+            type: string
+  /marc/{resourceId}/job/{username}/{timestamp}:
```

Comment:
Returns 200 if the conversion job is still proceeding. If completed, returns a 303 with a location header.

Reply:
Returns 500 if conversion job failed.

Reply:
The 500 is crucial, as a consumer needs to know when conversion failed.